### PR TITLE
[async-simple] Add new port async-simple

### PR DESCRIPTION
**Pull request by poor-circle**
- [✔ ] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [✔ ] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [✔ ] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [✔ ] The versioning scheme in `vcpkg.json` matches what upstream says.
- [✔ ] The license declaration in `vcpkg.json` matches what upstream says.
- [✔ ] The installed as the "copyright" file matches what upstream says.
- [✔ ] The source code of the component installed comes from an authoritative source.
- [✔] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [✔ ] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [✔ ] Only one version is in the new port's versions file.
- [✔ ] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by poor-circle** [comment]
> @poor-circle please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree [company="Alibaba"]

**Comment by poor-circle** [comment]
> @microsoft-github-policy-service agree company="your company"

@microsoft-github-policy-service agree company="Alibaba"

**Comment by poor-circle** [comment]
@microsoft-github-policy-service agree company="Alibaba"

File: ports/async-simple/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,28 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO alibaba/async_simple
+    REF "v1.3"
```
Comment (Cheney-W) [inline ports/async-simple/portfile.cmake:4]:
```suggestion
    REF "v${VERSION}"
```

File: ports/async-simple/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    SHA512 7686d8c98e5e87cf88c24488e4c0b528b29011ef48c25156c57695208dab75ee2f28b6f36e2612db2bf4644b50f4c1d4a7ebc54f16bc035cb106239d5b335c77
+    HEAD_REF main
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS -DASYNC_SIMPLE_ENABLE_TESTS=OFF -DASYNC_SIMPLE_BUILD_DEMO_EXAMPLE=OFF -DASYNC_SIMPLE_ENABLE_ASAN=OFF
```
Comment (Cheney-W) [inline ports/async-simple/portfile.cmake:11]:
```suggestion
    OPTIONS
        -DASYNC_SIMPLE_ENABLE_TESTS=OFF
        -DASYNC_SIMPLE_BUILD_DEMO_EXAMPLE=OFF
        -DASYNC_SIMPLE_ENABLE_ASAN=OFF
```

File: ports/async-simple/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+file(GLOB LIBS "${CURRENT_PACKAGES_DIR}/debug/lib/*async_simple*")
+
+list(LENGTH LIBS LIB_CNT)
+if (LIB_CNT EQUAL 0)
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
+endif()
+
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+configure_file("${CMAKE_CURRENT_LIST_DIR}/usage" "${CURRENT_PACKAGES_DIR}/share/${PORT}/usage" COPYONLY)
```
Comment (Cheney-W) [inline ports/async-simple/portfile.cmake:28]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
configure_file("${CMAKE_CURRENT_LIST_DIR}/usage" "${CURRENT_PACKAGES_DIR}/share/${PORT}/usage" COPYONLY)

```

**Comment by Cheney-W** [comment]
Usage test passed with x64-windows triplet.

**Review by data-queue — APPROVED** [review]

**Pull request merged**